src/engine/PiecewiseLinearConstraint.h

```diff
@@ -453,9 +453,21 @@ class PiecewiseLinearConstraint : public ITableau::VariableWatcher
         return _cdInfeasibleCases;
     }
 
-    void setTableauAuxVar( unsigned var )
+    /*
+      Add a variable to the list of aux vars designated in the Tableau
+      second argument is used for MaxConstraints
+    */
+    virtual void addTableauAuxVar( unsigned tableauAuxVar, unsigned /* constraintAuxVar */ )
```

Comment:
I'm guessing this is what you mentioned I would be commenting about.
Indeed, this is not how inheritance is usually applied - the parent can't know what MaxConstraints are.
If there's no other choice, keep it here, but make it more general: say that this value means X, and that children classes can use it if they need to.

Also, please explain to me why this parameter is needed, and why only for max.